src/App.test.jsx

```diff
@@ -0,0 +1,74 @@
+import { render, fireEvent } from '@testing-library/react';
+import App from './App';
+
+test('아무 것도 추가하지 않았을 때 초기 상태 확인', () => {
+  const { container, getByPlaceholderText } = render(
+    <App />,
+  );
+
+  const inputTodo = getByPlaceholderText('할 일을 입력해 주세요');
+
+  expect(container).toHaveTextContent('To-do');
+  expect(container).toHaveTextContent('할 일이 없어요!');
+  expect(inputTodo).toHaveAttribute('value', '');
+});
+
+test('Input에 할 일 입력 확인', () => {
+  const { getByPlaceholderText } = render(
+    <App />,
+  );
+
+  const inputTodo = getByPlaceholderText('할 일을 입력해 주세요');
+
+  fireEvent.change(inputTodo, {
+    target: {
+      value: '뭐라도 하기',
+    },
+  });
+
+  expect(inputTodo).toHaveAttribute('value', '뭐라도 하기');
+});
+
+test('<추가> 누르면 List에 추가 확인', () => {
+  const { container, getByText, getByPlaceholderText } = render(
+    <App />,
+  );
+
+  const addButton = getByText('추가');
+  const inputTodo = getByPlaceholderText('할 일을 입력해 주세요');
+
+  fireEvent.change(inputTodo, {
+    target: {
+      value: '아무 것도 하지 않기',
+    },
+  });
+
+  fireEvent.click(addButton);
```

Comment:
변수를 선언한 곳과 변수를 사용하는 곳은 가까이 위치하는게 좋습니다.

왜냐하면 변수에 대한 참조가 멀리 떨어져 있다면 코드를 읽는 사람이 프로그램 안에서 이리저리 옮겨 다녀야 합니다. 그리고 변수와 변수의 참조 사이에 있는 코드들은 `취약한 구간`입니다. 이 사이에서 변수에 어떤 일이 일어날지 파악하기 어렵기 때문입니다. 따라서 변수를 변하지 않는 상수로 사용하고, 변수에 대한 참조를 가까이 유지하면 프로그램을 좀 더 쉽게 이해할 수 있습니다.

이러한 것들은 변수가 선언된 곳과 참조된 곳의 범위를 계산하면 숫자로 측정도 가능합니다.

```suggestion
  const inputTodo = getByPlaceholderText('할 일을 입력해 주세요');
  fireEvent.change(inputTodo, {
    target: {
      value: '아무 것도 하지 않기',
    },
  });
  
  const addButton = getByText('추가');
  fireEvent.click(addButton);
```
See also
  - 코드 컴플리트2 - 10장 변수 사용시 고려할 것들 > 10.4 범위

Reply:
넵!! 변수를 선언하고 가까운 곳에서 참조할 수 있도록 습관 들이겠습니다 :)